shared/src/metabase/shared/util.cljc

```diff
@@ -12,3 +15,35 @@
                         (namespace k))]
       (str namespac "/" (name k))
       (name k))))
+
+(defn remove-nils
+  "Given a map, returns a new map with all nil values removed."
+  [m]
+  (m/filter-vals some? m))
+
+(defn normalize-key
+  "Given a string or keyword, in snake_case or kebab-case, normalize it to a :kebab-case keyword.
+
+  If the input is a namespaced keyword, the namespace is untouched but the base name is converted."
+  [k]
+  (keyword (when (keyword? k) (namespace k))
```

Comment:
Why not normalize the namespace too? Does that seem surprising or potentially dangerous to you? (I understand that in the currently envisioned cases this won't happen anyway.)

Reply:
I'm kind of assuming that if we're getting passed Clojure data structures they'll be named in Clojure style as `:some.name-space/foo`.